src/components/Divider.tsx

```diff
@@ -1,70 +1,182 @@
 import fastClick from '../util/fastClick'
 import head from '../util/head'
+import isDivider from '../util/isDivider'
+
+/** Custom hook to fetch divider-related data from the state, using rootedParentOf instead of parentOf. */
+const useDividerData = (path: Path) => {
+  return useSelector(
+    (state: State) => {
+      const dividerId = head(path)
+      const parentPath = rootedParentOf(state, path)
+      const parentId = head(parentPath)
+      const grandParentPath = parentId ? rootedParentOf(state, parentPath) : []
+      const grandParentId = head(grandParentPath)
+      const children = parentId ? getAllChildrenAsThoughts(state, parentId) : []
+      const childrenWithoutDividers = children.filter(child => !isDivider(child.value))
+      const isOnlyChild = childrenWithoutDividers.length === 0
+      const isTableView =
+        (parentId && attributeEquals(state, parentId, '=view', 'Table')) ||
+        (grandParentId && attributeEquals(state, grandParentId, '=view', 'Table'))
+
+      let thoughtsAtSameDepth: { id: ThoughtId; value: string }[] = []
+
+      if (isTableView && isOnlyChild && grandParentId) {
+        const parentSiblings = getAllChildrenAsThoughts(state, grandParentId)
+        thoughtsAtSameDepth = parentSiblings.flatMap(parent => {
+          const childrenOfParent = parent.id ? getAllChildrenAsThoughts(state, parent.id) : []
+          return childrenOfParent.filter(child => !isDivider(child.value))
+        })
+      }
+
+      return {
+        dividerId,
+        isOnlyChild,
+        isTableView,
+        children,
+        thoughtsAtSameDepth,
+      }
+    },
+    (prev, next) => {
+      return (
+        prev.dividerId === next.dividerId &&
+        prev.isOnlyChild === next.isOnlyChild &&
+        prev.isTableView === next.isTableView &&
+        prev.children.length === next.children.length &&
+        prev.thoughtsAtSameDepth.length === next.thoughtsAtSameDepth.length
+      )
+    },
+  )
+}
+
+/**
+ * Calculates the width of the thought text, including trailing spaces, to ensure the divider width matches the visual representation.
+ */
+const getThoughtWidths = (thoughts: { id: ThoughtId }[]) => {
+  return thoughts.map(thought => {
+    const innerThoughtElement = document.querySelector(
+      `[aria-label="tree-node"][data-id="${thought.id}"] [aria-label="thought"]`,
+    )
+
+    // Use DOM widths for thoughts
+    return innerThoughtElement?.getBoundingClientRect().width ?? 0
+  })
+}
 
 /** A custom horizontal rule. */
 const Divider = ({ path, cssRaw }: { path: Path; cssRaw?: SystemStyleObject }) => {
-  const dividerSetWidth = React.createRef<HTMLInputElement>()
   const dispatch = useDispatch()
+  const dividerRef = useRef<HTMLDivElement>(null)
+  const [dividerWidth, setDividerWidth] = useState<number>(DIVIDER_MIN_WIDTH)
+  const { dividerId, isOnlyChild, isTableView, children, thoughtsAtSameDepth } = useDividerData(path)
+  const resizeObservers = useRef<ResizeObserver[]>([])
```

Comment:
If we can remove the redundant cleanup at the start of `observeResize`, we should also be able to avoid this `useRef`. Instead, we can declare/instantiate a new `resizeObservers` each time the `useEffect` hook is called and disconnect/cleanup in the returned callback function.